### ISSUE TITLE
[clang-tidy] `bugprone-exception-escape.TreatFunctionsWithoutSpecificationAsThrowing: OnlyUndefined` complains about unrelated non-throwing function

### ISSUE THREAD

**Issue by firewave** [llvm/llvm-project#192436]
```cpp
#include <string>

struct Location
{
    Location() = default;
};

struct IfCond
{
    explicit IfCond(const Location& location, const std::string &E)
        : location(location), E(E) {}
    Location location;
    std::string E; // remove this member for the warning to go away
};

void f()
{
    Location l;
    IfCond ifcond(l, ""); // disable for warning to go away
}
```

```
<source>:3:8: warning: an exception may be thrown in function '~Location' which should not throw exceptions [bugprone-exception-escape]
    3 | struct Location
      |        ^
<source>:3:8: note: frame #0: an exception of unknown type may be thrown in function '~Location' here
```

https://godbolt.org/z/GfG8raqx6

As `Location` does not throw the warning is misleading at best.

**Comment by firewave** [comment]
CC @zeyi2
